gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -221,6 +228,9 @@ public void run() {
   private void consume() {
     try {
       Iterator<KafkaConsumerRecord> itr = gobblinKafkaConsumerClient.consume();
+      if(!enableAutoCommit) {
```

Comment:
Shouldn't this block be called after the while block?

Reply:
I don't think it really makes a difference because both adding messages to the queue and offset commiting are async operations. But it will make sense to keep it here if we ever decide to commit synchronously. In that case we are making sure we are committing the previous offsets and not the ones from the current consume() call